astacus/common/utils.py

```diff
@@ -135,6 +136,41 @@ async def httpx_request(
         return None
 
 
+@contextlib.asynccontextmanager
+async def httpx_request_stream(
+    url: str | httpx.URL,
+    *,
+    caller: str,
+    method: str = "get",
+    timeout: float = 10.0,
+    ignore_status_code: bool = False,
+    **kw,
+) -> AsyncIterator[httpx.Response | None]:
+    """Wrapper for httpx.request which handles timeouts as non-exceptions,
+    and returns only valid results that we actually care about.
+    """
+    # TBD: may need to redact url in future, if we actually wind up
+    # using passwords in urls here.
+    logger.info("async-request %s %s by %s", method, url, caller)
+    async with httpx.AsyncClient() as client:
+        try:
+            async with client.stream(method, url, timeout=timeout, **kw) as r:
+                if not r.is_error or ignore_status_code:
+                    yield r
+                    return
+                await r.aread()
```

Comment:
This is a weird api, so aread sets the `text` field

Reply:
Yes, this looks weird.
That's probably because text is not async (async properties are not a thing), and they don't want to block unexpectedly while reading `text`, instead it raises and exception if `aread` was not already called.